### PR TITLE
[PVR][keymaps][guiinfo] Fix / improve subchannel numbers support

### PR DESCRIPTION
**Pull request by ksooo**
Support for (ATSC) subchannel numbers was very limited and buggy. This PR does not make things perfect, but heads into the right direction to complete the support.

Direct channel number input:

screenshot000

All changes are runtime-tested on macOS, latest Kodi master.

@Jalle19 do you want to do the code review?
@ronie headsup for the removed guiinfo labels (no todos for Estuary/Estouchy/Confluence).

### CODE REVIEW

File: xbmc/pvr/PVRGUIActions.cpp [hunk 1]
```diff
@@ -1591,37 +1591,36 @@ namespace PVR
 
   void CPVRChannelSwitchingInputHandler::OnInputDone()
   {
-    int iChannelNumber;
     bool bSwitchToPreviousChannel;
-    {
-      CSingleLock lock(m_mutex);
```
Comment (Jalle19) [inline xbmc/pvr/PVRGUIActions.cpp]: Not needed anymore?

**Review by Jalle19 — APPROVED** [review]

**Comment by Jalle19** [comment]
Definitely a change for the better

File: xbmc/pvr/PVRGUIActions.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/pvr/PVRGUIActions.cpp]: Ooops. Good catch.

File: xbmc/pvr/channels/PVRChannelGroup.cpp [hunk 2]
```diff
@@ -424,15 +381,24 @@ bool CPVRChannelGroup::UpdateClientPriorities()
...
       CPVRChannelGroup::EmptyMember;
 }
 
+const PVRChannelGroupMember& CPVRChannelGroup::GetByUniqueID(const std::pair<int, int>& id) const
+{
+  CSingleLock lock(m_critSection);
+  auto it = m_members.find(id);
+  return it != m_members.end() ?
+  it->second :
+  CPVRChannelGroup::EmptyMember;
```
Comment (xhaggi) [inline xbmc/pvr/channels/PVRChannelGroup.cpp]: formatting seams wrong to me should be 
```
return it != m_members.end() 
  ? it->second 
  : CPVRChannelGroup::EmptyMember;
```

**Comment by xhaggi** [comment]
Looks good +1

File: xbmc/pvr/channels/PVRChannelGroup.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/pvr/channels/PVRChannelGroup.cpp]: Good catch. Damn auto indentation. Will change this ofc.

**Comment by ksooo** [comment]
Jenkins LINUX-64-Wayland testsuite crash on exit seems unrelated.

**Comment by ksooo** [comment]
For the PR itself, only Linux 64 wayland was failing, as I wrote...

**Comment by Rechi** [comment]
Linux 64 wayland is the one which failed for every build
never said all platforms are failing

File: xbmc/pvr/PVRChannelNumberInputHandler.cpp [hunk 3]
```diff
@@ -44,54 +46,69 @@ void CPVRChannelNumberInputHandler::OnTimeout()
...
 
-  for (std::deque<int>::const_reverse_iterator it = m_digits.rbegin(); it != m_digits.rend(); ++it)
+  size_t pos = m_inputBuffer.find(CPVRChannelNumber::SEPARATOR);
+  if (pos != std::string::npos)
+  {
+    // main + sub
+    if (pos != 0)
+    {
+      iChannelNumber = std::atoi(m_inputBuffer.substr(0, pos).c_str());
+      if (pos != m_inputBuffer.back())
```
Comment (Rechi) [inline xbmc/pvr/PVRChannelNumberInputHandler.cpp]: @ksooo What is this comparison good for?
Comparing pos (size_t) with the last element of the string (char) doesn't seem valid.

Comment (ksooo) [reply xbmc/pvr/PVRChannelNumberInputHandler.cpp]: Ooops. I will fix this.

**Pull request merged**